packages/shared/src/components/notifications/EnableNotification.tsx

```diff
@@ -39,7 +47,14 @@ function EnableNotification({
   source = NotificationPromptSource.NotificationList,
   parentCommentAuthorName,
 }: EnableNotificationProps): ReactElement {
+  const isExtension = !!process.env.TARGET_BROWSER;
   const [isEnabled, setIsEnabled] = useState(false);
+  const [permissionCache, setPermissionCache] =
```

Comment:
I think it will be better to add all the new logic here to the notifications context. I think the context can encapsulate it better, and then no matter when we want to use the logic it will already be there.
What do you think?

Reply:
I agree with that. It is better to encapsulate everything we can so we will only have to make changes in one place if necessary. Let me work on that quickly.